File: leetcode/852/Main.java
class Solution {
    public int peakIndexInMountainArray(int[] arr) {

        int left = 0;
        int right = arr.length -1;
        
        while(left < right)
        {
            int mid = left + (left+right) / 2; 
            if(arr[mid] < arr[mid+1])
            {
                left = mid+1;
            }else{
                right = mid;    
            }   
        }
        return left;
    }
}

class Main{
    public static void main(String[] args) {
        Solution sl = new Solution();
        System.out.println(sl.peakIndexInMountainArray(new int[]{1,2,3,2,1}));
    }
}

File: leetcode/852/Main.js
/**
 * @param {number[]} arr
 * @return {number}
 */
 var peakIndexInMountainArray = function(arr) {
    let left = 0;
    let right = arr.length -1;
    
    while(left < right)
    {
        
        let mid = left + parseInt((left+right) / 2); 
        
        console.log("before : ", left, right, mid);

        console.log("arr[mid] , arr[mid+1] : ", arr[mid] , arr[mid+1]);

        if(arr[mid] < arr[mid+1])
        {
            left = mid+1;
            // console.log("left : " + left);
        }else{
            right = mid;    
            // console.log("right : " + right);
        }   

        console.log("after : ", left, right, mid);
    }
    return left;
};

console.log(peakIndexInMountainArray([0,1,0]));